### PR TITLE
remove nodekeys from flex tree

### PR DESCRIPTION
**Pull request by daesunp**
## Description

This PR moves `NodeKeys` to the simple-tree layer, and replaces the use of `NodeKeys` with the `nodeKeyManager`

### CODE REVIEW

File: packages/dds/tree/src/feature-libraries/flex-tree/index.ts [hunk 1]
```diff
@@ -61,6 +61,6 @@ export {
 	FlexibleNodeSubSequence,
 } from "./flexTreeTypes.js";
 
-export { NodeKeys } from "./nodeKeys.js";
+export { NodeKeys } from "../../simple-tree/nodeKeys.js";
```
Comment (CraigMacomber) [inline packages/dds/tree/src/feature-libraries/flex-tree/index.ts:64]: this should be removed: flex tree should not depend on simple-tree.

File: packages/dds/tree/src/simple-tree/nodeKeys.ts [hunk 2]
```diff
@@ -3,9 +3,16 @@
  * Licensed under the MIT License.
  */
 
-import { LocalNodeKey, NodeKeyIndex, NodeKeyManager, StableNodeKey } from "../node-key/index.js";
+import {
+	LocalNodeKey,
+	NodeKeyIndex,
+	NodeKeyManager,
+	StableNodeKey,
+	// eslint-disable-next-line import/no-internal-modules
```
Comment (CraigMacomber) [inline packages/dds/tree/src/simple-tree/nodeKeys.ts:11]: this code should not need to depend on internals of other libraries. Wither import these from somewhere thats allowed (feature-libraries)

File: packages/dds/tree/src/simple-tree/nodeKeys.ts [hunk 3]
```diff
@@ -3,9 +3,16 @@
...
+} from "../feature-libraries/node-key/index.js";
 
-import { FlexTreeObjectNode } from "./flexTreeTypes.js";
+// eslint-disable-next-line import/no-internal-modules
+import { FlexTreeObjectNode } from "../feature-libraries/flex-tree/flexTreeTypes.js";
```
Comment (CraigMacomber) [inline packages/dds/tree/src/simple-tree/nodeKeys.ts:15]: If this is going to be at the simple tree layer, you shouldn't map to flex tree nodes. Prabably use TreeNode instead.

File: packages/dds/tree/src/simple-tree/nodeKeys.ts (hunk 2 above)
Comment (daesunp) [reply packages/dds/tree/src/simple-tree/nodeKeys.ts]: Updated illegal imports

File: packages/dds/tree/src/feature-libraries/flex-tree/index.ts (hunk 1 above)
Comment (daesunp) [reply packages/dds/tree/src/feature-libraries/flex-tree/index.ts:64]: Removed unnecessary export

File: packages/dds/tree/src/simple-tree/nodeKeys.ts (hunk 3 above)
Comment (daesunp) [reply packages/dds/tree/src/simple-tree/nodeKeys.ts]: Updated to use `TreeNode`

**Review by CraigMacomber — APPROVED** [review]

**Pull request merged**